apps/reaper/test/unit/reaper/data_extract/extract_step_test.exs

```diff
@@ -0,0 +1,538 @@
+            body: %{},
+            headers: %{},
+            cacheTtl: nil
+          },
+          assigns: %{
+            path: "fancyurl"
+          }
+        }
+      ]
+
+      assigns = ExtractStep.execute_extract_steps(dataset, steps)
+
+      assert assigns == %{path: "fancyurl", token: "auth_token"}
+    end
+
+    test "fails with a reasonable error message", %{bypass: bypass, dataset: dataset} do
+      Bypass.stub(bypass, "POST", "/", fn conn ->
+        Plug.Conn.resp(conn, 403, "No dice")
+      end)
+
+      steps = [
+        %{
+          type: "auth",
+          context: %{
+            path: ["sub", "path"],
+            destination: "token",
+            url: "http://localhost:#{bypass.port}",
+            encodeMethod: "json",
+            body: %{Key: "AuthToken"},
+            headers: %{},
+            cacheTtl: nil
+          },
+          assigns: %{}
+        }
+      ]
+
+      assert_raise RuntimeError, "Unable to process auth step for dataset 12345-6789.", fn ->
+        ExtractStep.execute_extract_steps(dataset, steps)
+      end
+    end
+  end
+
+  describe "execute_extract_steps/2 date" do
+    test "puts current date with format into assigns block", %{dataset: dataset} do
+      allow Timex.now(), return: DateTime.from_naive!(~N[2020-08-31 13:26:08.003], "Etc/UTC")
+
+      steps = [
+        %{
+          type: "date",
+          context: %{
+            destination: "currentDate",
+            deltaTimeUnit: nil,
+            deltaTimeValue: nil,
+            timeZone: nil,
+            format: "{YYYY}-{0M}"
+          },
+          assigns: %{}
+        }
+      ]
+
+      assert ExtractStep.execute_extract_steps(dataset, steps) ==
+               %{
+                 currentDate: "2020-08"
+               }
+    end
+
+    test "puts current date can do time delta", %{dataset: dataset} do
+      allow Timex.now(), return: DateTime.from_naive!(~N[2020-08-31 13:30:00.000], "Etc/UTC")
+
+      steps = [
+        %{
+          type: "date",
+          context: %{
+            destination: "currentDate",
+            deltaTimeUnit: "years",
+            deltaTimeValue: -33,
+            timeZone: nil,
+            format: "{YYYY}-{0M}"
+          },
+          assigns: %{}
+        }
+      ]
+
+      assert ExtractStep.execute_extract_steps(dataset, steps) ==
+               %{
+                 currentDate: "1987-08"
+               }
+
+      steps = [
+        %{
+          type: "date",
+          context: %{
+            destination: "currentDate",
+            deltaTimeUnit: "minutes",
+            deltaTimeValue: 33,
+            timeZone: nil,
+            format: "{YYYY}-{0M}-{0D} {h12}:{m}"
+          },
+          assigns: %{}
+        }
+      ]
+
+      assert ExtractStep.execute_extract_steps(dataset, steps) ==
+               %{
+                 currentDate: "2020-08-31 2:03"
+               }
+    end
+  end
+
+  describe "execute_extract_steps/2 secret" do
```

Comment:
Same with at least a failing case here or in a unit tests

Reply:
you found this below